custom/icds/management/commands/get_icds_sms_usage.py

```diff
@@ -119,25 +137,62 @@ def handle(self, domain, start_date, end_date, **options):
         ):
             location = self.get_location(sms)
             state_code = self.get_state_code(location)
-            if state_code not in data:
-                data[state_code] = {}
+            district_code = self.get_district_code(location)
+            if state_code not in district_level_data:
+                district_level_data[state_code] = {}
+            if district_code not in district_level_data[state_code]:
+                district_level_data[state_code][district_code] = {}
 
             indicator_slug = self.get_indicator_slug(sms)
-            if indicator_slug not in data[state_code]:
-                data[state_code][indicator_slug] = 0
-
-            data[state_code][indicator_slug] += 1
-
-        with open(filename, 'wb') as f:
-            headers = ('State Code', 'State Name', 'Indicator', 'SMS Count')
-            excel_data = []
+            if indicator_slug not in district_level_data[state_code][district_code]:
+                district_level_data[state_code][district_code][indicator_slug] = 0
+
+            district_level_data[state_code][district_code][indicator_slug] += 1
+
+        for state_code, state_data in district_level_data.items():
+            if state_code not in state_level_data:
+                state_level_data[state_code] = {}
+            for district_data in state_data.values():
+                for indicator_slug, count in district_data.items():
+                    if indicator_slug not in state_level_data[state_code]:
+                        state_level_data[state_code][indicator_slug] = 0
+                    state_level_data[state_code][indicator_slug] += count
+
+        with open(filename, 'wb') as excel_file:
+            state_headers = ('State Code', 'State Name', 'Indicator', 'SMS Count')
+            district_headers = (
+                'State Code', 'State Name', 'District Code', 'District Name', 'Indicator', 'SMS Count'
+            )
+            excel_state_data = []
+            excel_district_data = []
 
-            for state_code, state_data in data.items():
+            for state_code, state_data in state_level_data.items():
```

Comment:
I see the old report was not sorted, but it seems like it would be easier to review if this were sorted.
```suggestion
            for state_code, state_data in sorted(state_level_data.items()):
```